simple/app.go

```diff
@@ -0,0 +1,334 @@
+	kinds := make([]resource.Kind, 0)
+	for _, k := range a.kinds {
+		kinds = append(kinds, k.Kind)
+	}
+	return kinds
+}
+
+// Runner returns a resource.Runnable() that runs the underlying operator.InformerController and all custom runners
+// added via AddRunnable. The returned resource.Runnable also implements metrics.Provider, allowing the caller
+// to gather prometheus.Collector objects used by all underlying runners.
+func (a *App) Runner() app.Runnable {
+	return a.operator
+}
+
+// AddRunnable adds an arbitrary resource.Runnable runner to the App, which will be encapsulated as part of Runner().Run().
+// If the provided runner also implements metrics.Provider, PrometheusCollectors() will be called when called on Runner().
+func (a *App) AddRunnable(runner app.Runnable) {
+	a.operator.AddController(runner)
+}
+
+// ManageKind introduces a new kind to manage.
+func (a *App) ManageKind(kind AppManagedKind) error {
+	a.kinds[gvk(kind.Kind.Group(), kind.Kind.Version(), kind.Kind.Kind())] = kind
+	if kind.Reconciler != nil || kind.Watcher != nil {
+		client, err := a.clientGenerator.ClientFor(kind.Kind)
+		if err != nil {
+			return err
+		}
+		inf, err := operator.NewKubernetesBasedInformerWithFilters(kind.Kind, client, operator.ListWatchOptions{
+			Namespace:      kind.ReconcileOptions.Namespace,
+			LabelFilters:   kind.ReconcileOptions.LabelFilters,
+			FieldSelectors: kind.ReconcileOptions.FieldSelectors,
+		})
+		if err != nil {
+			return err
+		}
+		err = a.informerController.AddInformer(inf, kind.Kind.GroupVersionKind().String())
+		if err != nil {
+			return fmt.Errorf("could not add informer to controller: %v", err)
+		}
+		if kind.Reconciler != nil {
+			reconciler := kind.Reconciler
+			if !kind.ReconcileOptions.UsePlain {
+				op, err := operator.NewOpinionatedReconciler(client, a.getFinalizer(kind.Kind))
+				if err != nil {
+					return err
+				}
+				op.Wrap(kind.Reconciler)
+				reconciler = op
+			}
+			err = a.informerController.AddReconciler(reconciler, kind.Kind.GroupVersionKind().String())
+			if err != nil {
+				return fmt.Errorf("could not add reconciler to controller: %v", err)
+			}
+		}
+		if kind.Watcher != nil {
+			watcher := kind.Watcher
+			if !kind.ReconcileOptions.UsePlain {
+				op, err := operator.NewOpinionatedWatcherWithFinalizer(kind.Kind, client, a.getFinalizer)
+				if err != nil {
+					return err
+				}
+				if cast, ok := kind.Watcher.(syncWatcher); ok {
+					op.Wrap(cast, false)
+					op.SyncFunc = cast.Sync
+				} else {
+					op.Wrap(kind.Watcher, true)
+				}
+				watcher = op
+			}
+			err = a.informerController.AddWatcher(watcher, kind.Kind.GroupVersionKind().String())
+			if err != nil {
+				return fmt.Errorf("could not add watcher to controller: %v", err)
+			}
+		}
+	}
+	return nil
+}
+
+// RegisterKindConverter adds a converter for a GroupKind, which will then be processed on Convert calls
+func (a *App) RegisterKindConverter(groupKind schema.GroupKind, converter k8s.Converter) {
```

Comment:
Same as above – why would this be used over passing all converters upon initialization?